bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/SurePetcareAPIHelper.java

```diff
@@ -0,0 +1,474 @@
+        setDataThroughApi(ctrlurl, HttpMethod.PUT, control);
+
+        // now we're fetching the new state back for the cache
+        String devurl = DEVICE_BASE_URL + "/" + device.id.toString() + "/control";
+        SurePetcareDeviceControl newControl = SurePetcareConstants.GSON.fromJson(getDataFromApi(devurl),
+                SurePetcareDeviceControl.class);
+        if (newControl != null) {
+            newControl.curfewList = newControl.curfewList.order();
+        }
+        device.control = newControl;
+    }
+
+    /**
+     * Returns a unique device id used during the authentication process with the Sure Petcare API. The id is derived
+     * from the local MAC address or hostname.
+     *
+     * @return a unique device id
+     */
+    public final Integer getDeviceId() {
+        try {
+            return getDeviceId(NetworkInterface.getNetworkInterfaces(), InetAddress.getLocalHost());
+        } catch (UnknownHostException | SocketException e) {
+            logger.warn("unable to discover mac or hostname, assigning default device id {}", DEFAULT_DEVICE_ID);
+            return DEFAULT_DEVICE_ID;
+        }
+    }
+
+    /**
+     * Returns a unique device id used during the authentication process with the Sure Petcare API. The id is derived
+     * from the local MAC address or hostname provided as arguments
+     *
+     * @param interfaces a list of interface of this host
+     * @param localHostAddress the ip address of the localhost
+     * @return a unique device id
+     */
+    public final int getDeviceId(Enumeration<NetworkInterface> interfaces, InetAddress localHostAddress) {
+        int decimal = DEFAULT_DEVICE_ID;
+        try {
+            if (interfaces.hasMoreElements()) {
+                NetworkInterface netif = interfaces.nextElement();
+
+                byte[] mac = netif.getHardwareAddress();
+                if (mac != null) {
+                    StringBuilder sb = new StringBuilder();
+                    for (int i = 0; i < mac.length; i++) {
+                        sb.append(String.format("%02x", mac[i]));
+                    }
+                    String hex = sb.toString();
+                    decimal = Math.abs((int) (Long.parseUnsignedLong(hex, 16) % Integer.MAX_VALUE));
+                    logger.debug("current MAC address: {}, device id: {}", hex, decimal);
+                } else {
+                    String hostname = localHostAddress.getHostName();
+                    decimal = hostname.hashCode();
+                    logger.debug("current hostname: {}, device id: {}", hostname, decimal);
+                }
+            } else {
+                String hostname = localHostAddress.getHostName();
+                decimal = hostname.hashCode();
+                logger.debug("current hostname: {}, device id: {}", hostname, decimal);
+            }
+        } catch (SocketException e) {
+            logger.debug("Socket Exception: cause={}. message={}", e.getCause(), e.getMessage());
```

Comment:
A cause is a throwable i would expect it to appear as the last argument without a placeholder.

Reply:
done